From 99aec605daac02d214f1e5be77108210287dc533 Mon Sep 17 00:00:00 2001
From: Oliver Schmidt <os@flyingcircus.io>
Date: Fri, 21 Mar 2025 11:40:00 +0100
Subject: [PATCH] Use canonical network interface names from the kernel.

Under Linux it's possible for network interfaces to have alias
names (or "altnames"), added by "ip link property add
altname". However, /proc/net/dev only lists interfaces with their
canonical names, which means that dstat will reject alias names as
unknown. Instead, use if_nametoindex(3) and if_indextoname(3) to
discover the canonical names of the user-specified interfaces.

Original patch by Molly Miller <mm@flyingcircus.io> for dstat.
---
 dool | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/dool b/dool
index d355314..7421514 100755
--- a/dool
+++ b/dool
@@ -36,6 +36,7 @@ import os
 import re
 import resource
 import sched
+import socket
 import sys
 import time
 import signal
@@ -1270,7 +1271,14 @@ class dool_net(dool):
     def vars(self):
         ret = []
         if op.netlist:
-            varlist = op.netlist
+            varlist = []
+            for iface in op.netlist:
+                try:
+                    idx = socket.if_nametoindex(iface)
+                    name = socket.if_indextoname(idx)
+                    varlist.append(name)
+                except OSError:
+                    pass
         elif not op.full:
             varlist = ('total',)
         else:
-- 
2.47.2

